### PR TITLE
add tranco api shit

### PR DESCRIPTION
**Pull request by feyoav**

### CODE REVIEW

File: tranco_api.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        raise TrancoDbNotInitilizedError('DB is not initialized')
+
+    def get_top_sites_list(self) -> List[str]:
+        if self._top_sitest_list is not None:
+            return self._top_sitest_list
+        raise TrancoDbNotInitilizedError('DB is not initialized')
+
+    def set_db(self) -> None:
+        new_tranco_cache_dir = self._TRANCO_BASE_CACHE_DIR + \
+            str(datetime.datetime.now())
```
Comment (YotamShvartsun) [inline tranco_api.py]: No reason to use a new cache directory each time

File: tranco_api.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    
+    def is_site_safe(self, domain: str) -> bool:
+        return TRANCO_NOT_FOUND != self._latest_list.rank(domain)
+
+    def set_db(self) -> None:
+        if self._tranco_db is None:
+            self._tranco_db = Tranco(cache=True, cache_dir=self._TRANCO_BASE_CACHE_DIR)
+        self._latest_list = self._tranco_db.list()
+        new_top_sites_list = self._latest_list.top()
+        self._top_sitest_list = new_top_sites_list
```
Comment (feyoav) [inline tranco_api.py]:
```suggestion
        self._top_sitest_list = self._latest_list.top()
```

File: tranco_api.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        if self._tranco_db is not None:
+            return self._tranco_db
+        raise TrancoDbNotInitilizedError('DB is not initialized')
+
+    def get_top_sites_list(self) -> List[str]:
+        if self._top_sitest_list is not None:
+            return self._top_sitest_list
+        raise TrancoDbNotInitilizedError('DB is not initialized')
+    
+    def is_site_safe(self, domain: str) -> bool:
```
Comment (feyoav) [inline tranco_api.py:37]: maybe `is_site_knowen` or something like that?
if a site is not in the top million sites worldwide it's not necessarily not safe

Comment (YotamShvartsun) [reply tranco_api.py:37]: @sivanarazi is working on an interface, which exports a function named `is_site_safe`, the function is named to match the soon-to-be-added API

**Pull request merged**